### PR TITLE
Saga pattern was implemented

### LINKED ISSUE #3: Implement Saga pattern with Kafka

### PR DESCRIPTION
**Pull request by vpetrovaa**

### CODE REVIEW

File: src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java [hunk 1]
```diff
@@ -34,12 +40,42 @@ public Mono<CarAggregate> handle(Car car) {
...
+                        return aggregateService.rent(event, number);
+                    } else {
+                        callConfirmOrDenyMethod(aggregateId, "deny");
+                        return aggregate;
+                    }
+                });
+        return aggregate;
+    }
+
+    public void callConfirmOrDenyMethod(String aggregateId, String value){
```
Comment (h1alexbel) [inline src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java]: expose methods only in interfaces

File: src/main/java/com/solvd/carsharing/event/impl/EventServiceImpl.java [hunk 2]
```diff
@@ -1,11 +1,9 @@
 package com.solvd.carsharing.event.impl;
 
 import com.solvd.carsharing.command.CreateCarCommand;
+import com.solvd.carsharing.command.RentCarCommand;
 import com.solvd.carsharing.command.UpdateNumberCommand;
-import com.solvd.carsharing.event.CarCreatedEvent;
-import com.solvd.carsharing.event.Event;
-import com.solvd.carsharing.event.EventService;
-import com.solvd.carsharing.event.NumberUpdatedEvent;
+import com.solvd.carsharing.event.*;
```
Comment (h1alexbel) [inline src/main/java/com/solvd/carsharing/event/impl/EventServiceImpl.java]: don't use wildcard import

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java (hunk 1 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java]: @h1alexbel, you mean to add it to interface? I add this method to make my main handle method smaller, so I declared only handle methods in the interface

File: src/main/java/com/solvd/carsharing/event/impl/EventServiceImpl.java (hunk 2 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/carsharing/event/impl/EventServiceImpl.java]: https://github.com/vpetrovaa/carsharing/pull/4/commits/66b0ac6217d622d261e4c894e5a09def9beb04c9 fixed

File: src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java (hunk 1 above)
Comment (h1alexbel) [reply src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java]: @vpetrovaa since then, create a new class with `callConfirmOrDenyMethod`

**Review by h1alexbel — COMMENTED** [review]
@vpetrovaa extract `callConfirmOrDenyMethod ` logic to new class

File: src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java (hunk 1 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/carsharing/command/impl/CommandServiceImpl.java]: @h1alexbel, https://github.com/vpetrovaa/carsharing/pull/4/commits/3b1b1b937d2784c7ee2c08816edb0ac9c702dda9 fixed

File: src/main/java/com/solvd/carsharing/command/util/CommandUtilsImpl.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Service
+@RequiredArgsConstructor
+public class CommandUtilsImpl implements CommandUtils {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/carsharing/command/util/CommandUtilsImpl.java]: @vpetrovaa if it's utility class then it should be final class without a chance to construct it

if it's object that can perform some functionality -> should be named properly

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/solvd/carsharing/command/util/CommandUtilsImpl.java (hunk 3 above)
Comment (vpetrovaa) [reply src/main/java/com/solvd/carsharing/command/util/CommandUtilsImpl.java]: @h1alexbel, ok, this method is connected with access to another microservice. Is RentalService the correct name? So this method changes the status of rental https://github.com/vpetrovaa/carsharing/pull/4/commits/1d6b4e55308e74d734f7618e6bad13d1633ef852

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**